File: Easy/easy10.cpp
// reddit.com/r/dailyprogrammer/wiki/challenges
// 11/3/2014
// Prompt: Validate a phone number, as if written on an input form.
// Telephone numbers can be written as ten digits, or with dashes, spaces, or dots between the three segments, or with the area code parenthesized.
// White space between segments is optional.

#include <iostream>
#include <stdlib.h>
#include <string>
#include <stdio.h>
// Find the library for standard regex
// #include <regex>

int main(){
	std::string phoneNumber;
	bool isPhoneNumber = false;
	std::cout << "Please enter a phone number to be validated\n";
	std::cin >> phoneNumber;
	if (regex_match(phoneNumber, regex("[0-9]{3}-[0-9]{4}")))
		isPhoneNumber = true;
	if (std::regex_match(phoneNumber, regex("[0-9]{3}-[0-9]{3}-[0-9]{4}")))
		isPhoneNumber = true;
	// Match (xxx)xxx-xxx
	// Match (xxx)-xxx-xxx
	// Match xxx-xxx-xxx
	// Match xxx xxx xxx
	if (isPhoneNumber = true){
		std::cout << phoneNumber << " is a legitimate phone number.";
	}
	system("PAUSE");
	return 0;
}

File: Easy/easy10.py
# reddit.com/r/dailyprogrammer/wiki/challenges
# github.com/spokenlore
# 02/18/2015
# Prompt: Validate a phone number, as if written on an input form.
# Telephone numbers can be written as ten digits, or with dashes, spaces, or dots between the three segments, or with the area code parenthesized.
# White space between segments is optional.

import time
import re
import sys

def main():
	pattern1 = re.compile("\d\d\d\D\d\d\d\D\d\d\d\d")
	pattern2 = re.compile("\d\d\d\d\d\d\d\d\d\d")
	while (1):
		askInput = "Please enter a phone number. This program will tell you if the number is valid. If done checking valid phone numbers, enter 0.\n"
		telNumber = raw_input(askInput)

		if telNumber == 0:
			break
		else:
			if pattern1.match(telNumber) or pattern2.match(telNumber):
				print "%s is a valid phone number.\n" % (telNumber)
			else:
				print "%s is not a valid phone number.\n" % (telNumber)

	print telNumber

main()